### PR TITLE
fix(web): fixes illegal KMW event state - can't focus a null element

### LINKED ISSUE #11383: bug(web): `TypeError: Cannot read properties of null (reading 'tagName')`
Good reference event: https://keyman.sentry.io/issues/5262360279/events/c773d3ce356344d0809ea3296b7151fb/

Sentry Issue: [KEYMANWEB-COM-1K5](https://keyman.sentry.io/issues/5262360279/?referrer=github_integration)

```
TypeError: Cannot read properties of null (reading 'tagName')
  at this.focusControlEvent (https://s.keyman.com/kmw/engine/17.0.319/src/web/src/app/ui/kmwuitoolbar.ts:1061:14)
  at LegacyEventEmitter.callEvent (https://s.keyman.com/kmw/engine/17.0.319/src/web/src/engine/events/src/legacyEventEmitter.ts:142:16)
  at ContextManager.setActiveTarget (https://s.keyman.com/kmw/engine/17.0.319/src/web/src/app/browser/src/contextManager.ts:304:22)
  at ContextManager.forgetActiveTarget (https://s.keyman.com/kmw/engine/17.0.319/src/web/src/app/browser/src/contextManager.ts:222:10)
  at Ct.<anonymous> (https://s.keyman.com/kmw/engine/17.0.319/src/web/src/app/browser/src/contextManager.ts:167:16)
...
(8 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #11383.

@keymanapp-test-bot skip

### CODE REVIEW

**Review by mcdurdin — COMMENTED** [review]
Conceptually I think I understand this change but there are a couple of rough edges

File: web/src/app/browser/src/contextManager.ts [hunk 1]
```diff
@@ -299,12 +299,28 @@ export default class ContextManager extends ContextManagerBase<BrowserConfigurat
...
 
+    //Execute external (UI) code needed on focus if required
     if(sendEvents) {
-      // //Execute external (UI) code needed on focus if required
-      this.apiEvents.callEvent('controlfocused', {
-        target: target?.getElement() || null,
-        activeControl: previousTarget?.getElement()
-      });
+      const blurredElement = previousTarget?.getElement();
+      const focusedElement = target?.getElement();
```
[the remark this answers is not in the document]
Comment (jahorton) [reply web/src/app/browser/src/contextManager.ts:305]: I did miss that it's a match for `Ltarg`.

----

As for the block you quoted, that's about maintaining proper connection between the active element and any control-specific keyboard settings.  If focus was lost - like when the user interacts with a UI module - and then regained while the "restoring focus" mode is on, that indicates that the user just directly adjusted the keyboard settings for the current control.  `_BlurKeyboardSettings` saves those settings, while `_FocusKeyboardSettings` restores them.  

Admittedly, those details - what the two functions represent - are a case of "headspace documentation", I suppose - the two functions were originally refactored our from general control focus and control blur handling.  The names made sense then, but probably do make less sense now in their present form.

So, the quoted block here is about handling of keyboard settings, which is independent from detecting and updating _which_ control is focused.  They are a bit related though - if restoring focus, we're explicitly re-focusing an element that should already be considered active.  If we're not, we could be changing focus in a number of different ways.

**Comment by mcdurdin** [comment]
I notice the web build failed. Haven't investigated why...

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.322-beta](https://keyman.com/downloads/releases/beta/17.0.322)

**Pull request merged**